packages/driver/src/cypress/runner.ts

```diff
@@ -1723,28 +1733,14 @@ export default {
         return _resumedAtTestIndex
       },
 
-      cleanupQueue (numTestsKeptInMemory) {
-        const cleanup = (queue) => {
-          if (queue.length > numTestsKeptInMemory) {
-            const test = queue.shift()
-
-            delete _testsQueueById[test.id]
-
-            _.each(RUNNABLE_LOGS, (logs) => {
-              return _.each(test[logs], (attrs) => {
-                // we know our attrs have been cleaned
-                // now, so lets store that
-                attrs._hasBeenCleanedUp = true
-
-                return LogUtils.reduceMemory(attrs)
-              })
-            })
-
-            return cleanup(queue)
-          }
-        }
-
-        return cleanup(_testsQueue)
+      cleanupQueue (_numTestsKeptInMemory) {
+        // this wasn't helpful since we store all tests in memory twice....
+        // must offload test state to the server to correctly leverage /
+        // store the defined number of tests in memory.
+        // TODO: EMILY - finishing making this work
```

Comment:
Do we have an issue open for this that we can link here and assign to you?

Reply:
This is definitely helpful and important for reducing memory usage.

I think there might be some understandable confusion since this config property is named `numTestKeptInMemory`. A more accurate name would be `numTestsToKeepSnapshotsFor`. It's not really about the tests kept in memory, but the snapshots those tests hold onto. Without offloading the tests to the server (which would be a good improvement), we need to keep all tests and their metadata in memory. That in itself shouldn't take up that much memory. They're fairly small objects. What _do_ take up a lot of memory are DOM snapshots. `LogUtils.reduceMemory(attrs)` is what removes the snapshots from memory by nulling them out from the logs within the test, allowing them to be garbage collected.

This change is keeping snapshots from being removed, so this will actually make memory consumption worse.

I tested this out with the following:

```js
const { _ } = Cypress

const getSnapshotsLength = () => {
  const tests = Cypress.runner.getTests()
  const snapshots = _.flatMap(tests, 'commands[0].snapshots')
  .filter((snapshot) => snapshot != null) // filter out snapshots that have been nulled out

  return snapshots.length
}

_.times(100).forEach((i) => {
  const num = i + 1

  it(`test #${num}`, () => {
    cy.log(`log #${num}`).then(() => {
      console.log(`# of snapshots after ${num} tests:`, getSnapshotsLength())
    })
  })
})
```

I temporarily added the `Cypress.runner.getTests()` method:

<img width="437" alt="Screen Shot 2022-09-12 at 11 26 07 AM" src="https://user-images.githubusercontent.com/1157043/189713838-98174f24-4b32-45ba-9445-c7355f0af45b.png">

On `develop`, the number of snapshots stops at 50 (it logs 51 because the test code runs before the snapshots gets cleared by the driver, but it does end up being 50).

<img width="260" alt="Screen Shot 2022-09-12 at 11 21 42 AM" src="https://user-images.githubusercontent.com/1157043/189713932-008d095f-4f29-4bc8-bb3a-46003c32da30.png">

On this branch, the number of snapshots keeps increasing with every test and every command.

<img width="258" alt="Screen Shot 2022-09-12 at 11 27 08 AM" src="https://user-images.githubusercontent.com/1157043/189713965-61f9964f-2e26-435e-9926-525a44e65b6e.png">

`LogUtils.reduceMemory(attrs)` needs to remain until we implement offloading tests to the server or it's going to make memory usage worse in the interim.

Reply:
@chrisbreiding interesting. Thanks for the insightful repo steps. This wasn't completely obvious, I pushed up some changes to maintain this behavior. Curious on your thoughts.